lib/propshaft/resolver/dynamic.rb

```diff
@@ -7,11 +7,24 @@ def initialize(load_path:, prefix:)
     end
 
     def resolve(logical_path)
-      if asset = load_path.find(logical_path)
+      asset(logical_path) do |asset|
         File.join prefix, asset.digested_path
-      else
-        raise Propshaft::MissingAssetError.new(logical_path)
       end
     end
+
+    def integrity(logical_path)
+      asset(logical_path) do |asset|
+        asset.integrity
+      end
+    end
+
+    private
+      def asset(logical_path)
```

Comment:
I think this form a little strange. Why not just do`find_asset(logical_path).integrity`, and then have `find_asset` raise if missing?

Reply:
Fixed. I thought it would be bad to have duplicate code like that.